lib/coverage.js

```diff
@@ -269,19 +269,41 @@ internals.instrument = function (filename) {
 
     let skipStart = 0;
     let segmentSkip = false;
+
+    // In addition to maintaining the current coverage bypass state, support
+    // comments that push that state to a stack, allowing transpiled code to
+    // skip coverage without interfering with a user-defined bypass block.
+    const skipStack = [];
     tree.comments.forEach((comment) => {
 
-        const directive = comment.value.match(/^\s*\$lab\:coverage\:(off|on)\$\s*$/);
+        const directive = comment.value.match(/^\s*\$lab\:coverage\:(off|on|push|pop)\$\s*$/);
         if (directive) {
-            const skip = directive[1] !== 'on';
-            if (skip !== segmentSkip) {
-                segmentSkip = skip;
-                if (skip) {
-                    skipStart = comment.range[1];
+            const command = directive[1];
```

Comment:
This whole block ends up being a bit of a gross deeply nested block. I could split bits out into functions or reorganize using `switch` or something; let me know if that's important - this seemed sufficient.